netCDF4/_netCDF4.pyx

```diff
@@ -3787,6 +3787,8 @@ rename a `netCDF4.Variable` attribute named `oldname` to `newname`."""
                     self._assign_vlen(elem, data)
                     return
                 elif data.dtype.kind in ['S', 'U']:
+                    if ma.isMA(data):
+                        data = data.filled(default_fillvals[data.dtype.kind])
```

Comment:
If the objective is just to avoid a crash here when a masked array is given (and not to provide full auto masking/unmasking support), why not just do

    if ma.isMA(data); data = data.filled()